### ISSUE TITLE
wrong-type-argument error when inserting "" in org-mode

### ISSUE THREAD

**Issue by shankar2k** [Fuco1/smartparens#994]
I have a problem when trying to insert double quotes (") in org-mode where, instead of generating a pair of quotes, a wrong-type-argument error is generated.
 
## Expected behavior

When I type a single `"` in org-mode, I expect smartparens to insert `""`, highlight them, and put the cursor in between the quotes

## Actual behavior

A single `"` is inserted and then I get the error message: `sp--post-self-insert-hook-handler: (wrong-type-argument char-or-string-p nil)`

## Steps to reproduce the problem

1. Open any org-mode document in Emacs
2. Press `"`

## Backtraces if necessary (`M-x toggle-debug-on-error`)
```
Debugger entered--Lisp error: (wrong-type-argument char-or-string-p nil)
  sp--escape-region(("\"") 29246 29247)
  sp-escape-open-delimiter()
  sp--post-self-insert-hook-handler()
  self-insert-command(1)
  org-self-insert-command(1)
  funcall-interactively(org-self-insert-command 1)
  call-interactively(org-self-insert-command nil nil)
  command-execute(org-self-insert-command)
```

## Environment & version information

```
- `smartparens` version: 1.11.0
- Active major-mode: org-mode (9.2.6)
- Emacs version (`M-x emacs-version`): GNU Emacs 26.2 (build 1, x86_64-w64-mingw32) of 2019-04-17
- Spacemacs/Evil/Other starterkit (specify which)/Vanilla: Vanilla
- OS: windows-nt
```

**Comment by shankar2k** [comment]
I just upgraded to GNU Emacs 26.3 (build 1, x86_64-w64-mingw32) of 2019-08-29, and the error persists

**Comment by notetiene** [comment]
I've noticed the same bug in my big configuration file.  The bug works with
`emacs -Q` (and loading smartparens).

I've simplified it so that the following code triggers the error:
```org
"\""
(insert double quote here→ )
...
"\""
(insert double quote here→ )
```

Here are the conditions when the bug happens:
 - it must be **org-mode**
 - a double quote `"` must be escaped (either in code blocks or text)
 - the cursor is inside a pair  (`[]`, `{}`, `<>`, ...)
 - the double quote character is typed

We can see that after another double quote is escaped, no error is issued after
that.

I hope it helps
